plugins/woocommerce-admin/client/marketplace/components/category-selector/category-dropdown.tsx

```diff
@@ -12,15 +14,38 @@ import { Category } from './category-selector';
 
 function DropdownContent( props: {
 	readonly categories: Category[];
+	readonly selected?: Category;
+	readonly onClick: () => void;
 } ): JSX.Element {
+	function updateCategorySelection(
+		event: React.MouseEvent< HTMLButtonElement >
+	) {
+		const slug = event.currentTarget.value;
+		props.onClick();
+
+		navigateTo( {
```

Comment:
This is awesome!

Reply:
Agreed. I was going to have to reach back out to the context and initiate a new API request. But `woocommerce/navigation` makes it super easy. This also enables browsers back button to go back to the previous state :100: